Remove issue 10116 note from collector docs (#3142)
open-telemetry/opentelemetry.io: content/en/docs/collector/_index.md

It is also easier to [setup a collector](./getting-started) than you might
think: the default OTLP exporters in each language assume a local collector

> endpoint, so you'd start up a collector and you'd start getting telemetry.

Reviewer: How about:

```suggestion
endpoint, so if you launch a collector it will automatically start receiving telemetry.
```

## Status and releases

The **collector** status is: [mixed][], since core collector components
currently have mixed [stability levels][].

> **Collector components** differ in their maturity levels. Every component has
> its stability documented in their respective `README.md`. You can find a list of
> all available collector components in the [registry][].

Reviewer: ```suggestion
**Collector components** differ in their maturity levels. Each component has
its stability documented in its `README.md`. You can find a list of
all available collector components in the [registry][].
```
